ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/assertion/mapper/ClaimantResponseMapperTest.java

```diff
@@ -31,82 +33,116 @@
+        assertNotNull(ccdResponse.getSubmittedOn());
+    }
 
-        //then
+    @Test
+    public void shouldMapClaimantAcceptanceWithClaimantPaymentIntentionBySetDateToCCDClaimantAcceptance() {
+        ClaimantResponse response = SampleClaimantResponse.ClaimantResponseAcceptation.builder()
+            .buildAcceptationIssueCCJWithClaimantPaymentIntentionBySetDate();
+        Claim claim =  claimBuilder.claimantResponse(response)
+            .claimantRespondedAt(LocalDateTimeFactory.nowInLocalZone())
+            .build();
+        CCDClaimantResponse ccdResponse = mapper.to(claim);
         assertThat((ResponseAcceptation) response).isEqualTo((CCDResponseAcceptation)ccdResponse);
+        assertNotNull(ccdResponse.getSubmittedOn());
     }
 
     @Test
-    public void shouldMapClaimantAcceptanceWithSettlementFormalisationFromCCD() {
-        //given
-        CCDClaimantResponse ccdResponse = SampleData.getResponseAcceptation(SETTLEMENT);
+    public void shouldMapClaimantAcceptanceWithCourtDeterminationPayByInstalmentsToCCDClaimantAcceptance() {
+        ClaimantResponse response = SampleClaimantResponse.ClaimantResponseAcceptation.builder()
+            .buildAcceptanceIssueSettlementWithCourtDeterminationPayByInstalments();
+        Claim claim =  claimBuilder.claimantResponse(response)
+            .claimantRespondedAt(LocalDateTimeFactory.nowInLocalZone())
+            .build();
+        CCDClaimantResponse ccdResponse = mapper.to(claim);
+        assertThat((ResponseAcceptation) response).isEqualTo((CCDResponseAcceptation)ccdResponse);
+        assertNotNull(ccdResponse.getSubmittedOn());
+    }
 
-        //when
-        ClaimantResponse response = mapper.from(ccdResponse).getClaimantResponse().get();
+    @Test
+    public void shouldMapClaimantAcceptanceWithClaimantPaymentIntentionPayImmediatelyToCCDClaimantAcceptance() {
+        ClaimantResponse response = SampleClaimantResponse.ClaimantResponseAcceptation.builder()
+            .buildAcceptanceIssueSettlementWithClaimantPaymentIntentionPayImmediately();
+        Claim claim =  claimBuilder.claimantResponse(response)
+            .claimantRespondedAt(LocalDateTimeFactory.nowInLocalZone())
+            .build();
+        CCDClaimantResponse ccdResponse = mapper.to(claim);
+        assertThat((ResponseAcceptation) response).isEqualTo((CCDResponseAcceptation)ccdResponse);
+        assertNotNull(ccdResponse.getSubmittedOn());
+    }
 
-        //then
-        assertThat((ResponseAcceptation)response).isEqualTo((CCDResponseAcceptation) ccdResponse);
+    @Test
+    public void shouldMapCCDResponseAcceptationWithCCJFormalisationToResponseAcceptation() {
+        CCDResponseAcceptation ccdResponse = SampleData.getResponseAcceptation(CCDFormaliseOption.CCJ);
+        mapper.from(ccdResponse,claimBuilder);
+        Claim claim = claimBuilder.build();
+        if (claim.getClaimantResponse().isPresent()) {
```

Comment:
this can be aligned with ifPresent

Reply:
I am not sure how that would be better. I will consider making this change if you can elaborate on why that is better.

Reply:
generally test shouldn't have if block for asserting, however in this scenario it has , so better to use ifPresent on Optional field as it is being followed across our code for consistency (internally they are same, however ifPresent looks cleaner).

Reply:
IMO for tests the better option would be:
```
 assertThat(claim).isNotNull();
 assertThat(claim.getClaimantResponse()).isPresent();
```

Reply:
@dharmendrak : can you please show me any standard documentation which validates your claim.
@sabahirfan  : IMO assertNotNull(claim.getClaimantRespondedAt().orElse(null)); does the same as assertThat(claim.getClaimantResponse()).isPresent()
assertThat(claim).isNotNull(); is an unnecessary assertion in this test case